src/credentials/tests/TestGroupDataProvider.cpp

```diff
@@ -582,6 +579,246 @@ void TestKeysIterator(nlTestSuite * apSuite, void * apContext)
+    NL_TEST_ASSERT(apSuite, CHIP_NO_ERROR == err);
+    NL_TEST_ASSERT(apSuite, keys1a.policy == keys_out.policy);
+    NL_TEST_ASSERT(apSuite, keys1a.num_keys_used == keys_out.num_keys_used);
+    NL_TEST_ASSERT(apSuite, !memcmp(keys1a.epoch_keys, keys_out.epoch_keys, sizeof(keys1a.epoch_keys[0]) * keys1a.num_keys_used));
+
+    err = groups->GetKeySet(kFabricIndex2, 505, keys_out);
+    NL_TEST_ASSERT(apSuite, CHIP_NO_ERROR == err);
+    NL_TEST_ASSERT(apSuite, keys0a.policy == keys_out.policy);
+    NL_TEST_ASSERT(apSuite, keys0a.num_keys_used == keys_out.num_keys_used);
+    NL_TEST_ASSERT(apSuite, !memcmp(keys0a.epoch_keys, keys_out.epoch_keys, sizeof(keys0a.epoch_keys[0]) * keys0a.num_keys_used));
+
+    err = groups->GetKeySet(kFabricIndex1, 202, keys_out);
+    NL_TEST_ASSERT(apSuite, CHIP_NO_ERROR == err);
+    NL_TEST_ASSERT(apSuite, keys0a.policy == keys_out.policy);
+    NL_TEST_ASSERT(apSuite, keys0a.num_keys_used == keys_out.num_keys_used);
+    NL_TEST_ASSERT(apSuite, !memcmp(keys0a.epoch_keys, keys_out.epoch_keys, sizeof(keys0a.epoch_keys[0]) * keys0a.num_keys_used));
+
+    err = groups->GetKeySet(kFabricIndex1, 404, keys_out);
+    NL_TEST_ASSERT(apSuite, CHIP_NO_ERROR == err);
+    NL_TEST_ASSERT(apSuite, keys1a.policy == keys_out.policy);
+    NL_TEST_ASSERT(apSuite, keys1a.num_keys_used == keys_out.num_keys_used);
+    NL_TEST_ASSERT(apSuite, !memcmp(keys1a.epoch_keys, keys_out.epoch_keys, sizeof(keys1a.epoch_keys[0]) * keys1a.num_keys_used));
+
+    err = groups->GetKeySet(kFabricIndex1, 606, keys_out);
+    NL_TEST_ASSERT(apSuite, CHIP_NO_ERROR == err);
+    NL_TEST_ASSERT(apSuite, keys0a.policy == keys_out.policy);
+    NL_TEST_ASSERT(apSuite, keys0a.num_keys_used == keys_out.num_keys_used);
+    NL_TEST_ASSERT(apSuite, !memcmp(keys0a.epoch_keys, keys_out.epoch_keys, sizeof(keys0a.epoch_keys[0]) * keys0a.num_keys_used));
+
+    // Remove Fabric
+    err = groups->RemoveFabric(kFabricIndex1);
+    NL_TEST_ASSERT(apSuite, CHIP_NO_ERROR == err);
+
+    // Mappings
+
+    exists = groups->GroupMappingExists(kFabricIndex1, group1a);
+    NL_TEST_ASSERT(apSuite, !exists);
+
+    exists = groups->GroupMappingExists(kFabricIndex1, group1b);
+    NL_TEST_ASSERT(apSuite, !exists);
+
+    exists = groups->GroupMappingExists(kFabricIndex1, group1c);
+    NL_TEST_ASSERT(apSuite, !exists);
+
+    exists = groups->GroupMappingExists(kFabricIndex2, group1a);
+    NL_TEST_ASSERT(apSuite, exists);
+
+    exists = groups->GroupMappingExists(kFabricIndex2, group1b);
+    NL_TEST_ASSERT(apSuite, !exists);
+
+    exists = groups->GroupMappingExists(kFabricIndex2, group1c);
+    NL_TEST_ASSERT(apSuite, exists);
+
+    // States
+
+    err = groups->GetGroupState(0, state0b);
+    NL_TEST_ASSERT(apSuite, CHIP_NO_ERROR == err);
+    NL_TEST_ASSERT(apSuite, state1a.fabric_index == state0b.fabric_index);
+    NL_TEST_ASSERT(apSuite, state1a.group == state0b.group);
+    NL_TEST_ASSERT(apSuite, state1a.key_set_index == state0b.key_set_index);
```

Comment:
It's not easy to understand, from the naming convention here, why those should match after RemoveFabric.